test/generate_tangent.jl

```diff
@@ -0,0 +1,37 @@
+using ChainRulesTestUtils: rand_tangent
+
+# Test struct for `rand_tangent`.
+struct Foo
+   a::Float64
+   b::Int
+   c::Any
+end
+
+@testset "generate_tangent" begin
+    rng = MersenneTwister(123456)
+
+    foreach([
+        ("hi", DoesNotExist),
+        ('a', DoesNotExist),
+        (:a, DoesNotExist),
+        (true, DoesNotExist),
+        (4, DoesNotExist),
+        (5.0, Float64),
+        (5.0 + 0.4im, Complex{Float64}),
+        (randn(Float32, 3), Vector{Float32}),
+        (randn(Complex{Float64}, 2), Vector{Complex{Float64}}),
+        (randn(5, 4), Matrix{Float64}),
+        (randn(Complex{Float32}, 5, 4), Matrix{Complex{Float32}}),
+        ([randn(5, 4), 4.0], Vector{Any}),
+        ((4.0, ), Composite{Tuple{Float64}}),
+        ((5.0, randn(3)), Composite{Tuple{Float64, Vector{Float64}}}),
+        ((a=4.0, ), Composite{NamedTuple{(:a,), Tuple{Float64}}}),
+        ((a=5.0, b=1), Composite{NamedTuple{(:a, :b), Tuple{Float64, Int}}}),
+        (sin, typeof(NO_FIELDS)),
+        (Foo(5.0, 4, rand(rng, 3)), Composite{Foo}),
```

Comment:
Nested structs are supported, right? Could you test something like `Foo(5.0, 4, Foo(2.0, 3, 'a'))`?